lineman/app/components/thread_page/comment_form/comment_form.coffee

```diff
@@ -3,15 +3,16 @@ angular.module('loomioApp').directive 'commentForm', ->
   restrict: 'E'
   templateUrl: 'generated/components/thread_page/comment_form/comment_form.html'
   replace: true
-  controller: ($scope, FlashService, Records, CurrentUser) ->
+  controller: ($scope, FormService, Records, CurrentUser) ->
     group = $scope.comment.discussion().group()
     discussion = $scope.comment.discussion()
 
-    $scope.submit = ->
-      $scope.comment.save().then ->
-        $scope.comment = Records.comments.build(discussion_id: discussion.id)
-        $scope.$emit('commentSaveSuccess')
-        FlashService.success('comment_form.messages.created')
+    $scope.submit = FormService.submit $scope, $scope.comment,
+      flashSuccess: 'comment_form.messages.created'
+      successCallback: =>
+        $scope.comment.body = ''
```

Comment:
FormService.submit does return a function, but I think it might need to
take the model as an argument, which I'll test out at some point

code is clearer; instead of

```
submit: (scope, model, options) ->
  ->
```

I think it needs to be
```
submit: (scope, options) ->
  (model) ->
```

Which involves changing all of the `ng-click: 'submit()' calls in our views
to take a model.


On Sun, Aug 2, 2015 at 2:44 AM, Robert Guthrie <notifications@github.com>
wrote:

> In lineman/app/components/thread_page/comment_form/comment_form.coffee
> <https://github.com/loomio/loomio/pull/2362#discussion_r36041330>:
>
> >      group = $scope.comment.discussion().group()
> >      discussion = $scope.comment.discussion()
> >
> > -    $scope.submit = ->
> > -      $scope.comment.save().then ->
> > -        $scope.comment = Records.comments.build(discussion_id: discussion.id)
> > -        $scope.$emit('commentSaveSuccess')
> > -        FlashService.success('comment_form.messages.created')
> > +    $scope.submit = FormService.submit $scope, $scope.comment,
> > +      flashSuccess: 'comment_form.messages.created'
> > +      successCallback: =>
> > +        $scope.comment.body = ''
>
> That's weird that the formservice holds state from the previous use. It
> seems to me that the $scope.comment reference is old because you need to
> wrap the FormService.submit in a function..
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2362/files#r36041330>.
>